tests/correctness/expressions.rs

```diff
@@ -222,3 +222,33 @@ fn enums_can_be_compared() {
     let _: i32 = compile_and_run(function, &mut main);
     assert_eq!([true, true, true], [main.a, main.b, main.c]);
 }
+
+#[test]
+fn amp_as_and_correctness_test() {
+    struct Main {
```

Comment:
The structs for the correctness tests need to have the exact representation of the program under test, that means in this case the struct should also contain a,b, and c. It is also preferable to have a repr(C) on it.
You can then grab a default copy of it to initialize by deriving default : 

```rust
#[derive(Default)]
#[repr(C)]
struct Main {
  a : bool,
  b : bool,
  c : bool,
  d : bool,
  e : bool,
}

let mut main = Main::default();
```

Reply:
thank you, will add this